app/models/ride.rb

```diff
@@ -1,3 +1,15 @@
 class Ride < ApplicationRecord
   belongs_to :amusement_park
+  has_many :mechanic_rides
+  has_many :mechanics, through: :mechanic_rides
+
+  validates :name, presence: true
+  validates :thrill_rating, presence: true
+  validates :open, inclusion: [true, false]
+
+  def avg_mechanic_experience
+    # binding.pry
+    # Ride.joins(:mechanics).distinct.select("avg(mechanics.years_experience) as avg_exp").where("rides.id = #{self.id}").avg_exp
+    Ride.joins(:mechanics).distinct.where("rides.id = #{self.id}").average("mechanics.years_experience")
+  end
```

Comment:
You were on the right track! You're right that you need a select to grab the ride name as well as the aggregate function the years of experience (in this case, it's average). Then you can group by ride. I encourage you to try it out FIRST before looking at the bottom of this comment where a sample query exists
.
.
.
.
.
.
.
.

```    rides.joins(:mechanics)
         .select("rides.name, AVG(mechanics.years_experience) AS avg_mech_exp")
         .group("rides.id")
         .order("avg_mech_exp DESC")
```